### PR TITLE
Fixes #32915 - accept strings as setting names

### PR DESCRIPTION
**Pull request by lzap**
I do not know why we use the super-weird :setting: "xyz" syntax but it is confusing. This is not a typical YAML. It is not a typical UNIX config. It's some kind of Ruby-YAML hybrid.

This patch converts all keys to symbols, we can do a followup puppet update and write all config files without the colons.

### CODE REVIEW

File: lib/proxy/plugin_initializer.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def configure
   end
 
   def fail_group(an_exception)
-    fail_group_with_message("Disabling all modules in the group #{printable_module_names(member_names)} due to a failure in one of them: #{an_exception}", an_exception.backtrace)
+    fail_group_with_message("Disabling all modules in the group #{printable_module_names(member_names)} due to a failure in one of them: #{an_exception}", an_exception)
```
Comment (lzap) [inline lib/proxy/plugin_initializer.rb]: When debugging this, backtrace was not printed correctly. It's because the method accepts exception, not backtrace array.

File: lib/proxy/settings.rb [hunk 2]
```diff
@@ -16,14 +16,14 @@ def self.initialize_global_settings(settings_path = nil, argv = ARGV)
   def self.load_plugin_settings(defaults, settings_file, settings_directory = nil)
     settings = {}
     begin
-      settings = YAML.load(File.read(File.join(settings_directory || ::Proxy::SETTINGS.settings_directory, settings_file))) || {}
+      settings = read_settings_file(settings_file, settings_directory)
```
Comment (lzap) [inline lib/proxy/settings.rb:19]: This line was just a copy of the line below, so DRY.

File: lib/proxy/plugin_initializer.rb [hunk 3]
```diff
@@ -114,7 +114,7 @@ def configure
   end
 
   def fail_group(an_exception)
-    fail_group_with_message("Disabling all modules in the group #{printable_module_names(member_names)} due to a failure in one of them: #{an_exception}", an_exception.backtrace)
```
Comment (ekohl) [inline lib/proxy/plugin_initializer.rb]: Would you mind submitting this as a separate PR? Looks like a good fix, but an unrelated fix.

**Review by ekohl — COMMENTED** [review]
I looked and couldn't find any settings that are a hash, but I wonder how that will happen. It'll be weird if only the first level accepts strings where the inner levels only accept symbols. Did you happen to check for this in plugins?

File: lib/proxy/plugin_initializer.rb (hunk 3 above)
Comment (lzap) [reply lib/proxy/plugin_initializer.rb]: Sure, created: https://projects.theforeman.org/issues/33482

Edit: Ooops this was fixed in b94c9b07

**Comment by lzap** [comment]
Rebased, @ekohl
